🍒 5.5 [Concurrency] SR-15309: Fix instance task.isCancelled impl to use _task

ktoso opened the pull request:
**Description:** The instance version of the `isCancelled` function on `Task` was incorrectly reaching for the _current task_ and checking that, rather than checking the `_task` it represents. The `UnsafeCurrentTask` version of this API was correct, only this API was mistaken.
**Risk:** Low
**Review by:** @DougGregor
**Testing:** Added test covering this case, PR testing on CI.
**Original PR:** https://github.com/apple/swift/pull/39710
**Radar:** rdar://84146091 & [SR-15309](https://bugs.swift.org/browse/SR-15309)

ktoso committed:
[Concurrency] SR-15309: Fix instance task.isCancelled impl to use _task

ktoso commented:
I checked other APIs while looking at this, no other APIs have this mistake.

ktoso commented:
@swift-ci please test

DougGregor commented on stdlib/public/Concurrency/TaskCancellation.swift:
```diff
@@ -51,13 +51,7 @@ extension Task {
   ///
   /// - SeeAlso: `checkCancellation()`
   public var isCancelled: Bool {
```
```suggestion
  @_transparent public var isCancelled: Bool {
```

Making this `@_transparent` will fix this more comprehensively: existing clients running on a newer OS will get the fix automatically (from the ABI-stable version), and clients recompiled with a newer _Concurrency module will inline the fixed version.

DougGregor committed:
Update stdlib/public/Concurrency/TaskCancellation.swift

Mark `isCancelled` as `@_transparent` so it will get inlined and fix the underling bug even for existing, shipped OS's.
